### DISCUSSION TITLE
TLS 1.2 minimum protocol

### DISCUSSION THREAD

**Discussion by davidlfox** [aws/aws-sdk-ruby#2827] [Q&A]
Per the [AWS changes coming in June](https://aws.amazon.com/blogs/security/tls-1-2-required-for-aws-endpoints/), how can I tell the oldest version of this SDK that supports TLS 1.2?

My app is using `'~> 1.96', '>= 1.96.1'`

I don't currently have any of the AWS-side logs configured (e.g. CloudWatch, etc) to determine from the server side. How can I confirm this from the client side?

**Comment by alextwoods** [comment]
See: https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/tls-version.html - basically, the version of openssl used determines the TLS 1.2 support.  You need OpenSSL 1.0.1 or higher - you can check with: 

```
ruby -r openssl -e 'puts OpenSSL::OPENSSL_VERSION'
```
